client/src/lib.rs

```diff
@@ -116,16 +116,23 @@ impl ApiKeyServiceClient {
     pub async fn make_request(
         &self,
         request: reqwest::Request,
+        api_url_extra: String,
```

Comment:
This assumes the the request coming from the user only has the base part of the URL.  I think we should assume it is the full url and omit this argument.